### ISSUE TITLE
[FR] Allow users to download series images or NFO data with any output template

### ISSUE THREAD

**Issue by ggttttggggtt** [kieraneglin/pinchflat#471]
**Is your feature request related to a problem? Please describe.**
Currently, channel posters and banners can only be downloaded if output is set to split by seasons. I don't use a media server but I want my archive to include channel images.

**Describe the solution you'd like**
Allowing this toggles with any output template. 

**Describe alternatives you've considered**
N/A

**Additional context**
There might also be people who want to download NFO data but would prefer not to split media into seasons. I think it would suffice to include a warning that media servers might not play well with a different output structure, in case someone has an unusual setup or just wants the data for some reason.

**Comment by kieraneglin** [comment]
Hey there! Thanks for the suggestion (:

This is something that seems easy for _most_ cases, but gets really tricky for the edge cases (which is why it's currently scoped to files split into seasons). The core of the issue is that the app needs to find a "root" directory in which to place the NFO/image files. For many normal cases this is easy - look for the directory that matches the source's custom name or the name of the channel/playlist. 

Where it gets tricky are the cases where either a root directory is shared among many sources and/or the output template doesn't use the name of the source/channel. Something like this:

```
/music/{{ genre }}/{{ artist }}/{{ title }}.{{ ext }}
```

This example would put media in different folders based on genre _and_ artist so what would the root directory be? Probably `/music/`, but then what happens when a completely different source uses this same output template? In this case there are many "right" answers but they all come with downsides or unexpected behaviours

---

That's not to say it's impossible, but I haven't come up with a good way to do it yet. It's one of those issues that sounds really simple on the face but gets weirder and weirder as you dig into it. I'm open to suggestions!

**Comment by ggttttggggtt** [comment]
I would find it acceptable for images to always go into the artist folder(s, but that's just me. I can't think of an elegant solution.

**Comment by kieraneglin** [comment]
Fair enough! But the problem then is that each media item would be separated based on genre. So if an artist has a rock song and a pop song, a rock and pop directory will be created and each of those directories would have a subdirectory for the artist. Would the metadata go in each of those artist directories? I think most people would say no but others might say yes.

I don't want to give the impression I'm throwing in the towel on this. I think there is an opportunity to cover more (but not all) cases, but I do have to be very thoughtful in how I approach this and where I draw the line

**Comment by ggttttggggtt** [comment]
Totally understandable, good things come with time
